chainstate/tx-verifier/src/transaction_verifier/reward_distribution.rs

```diff
@@ -299,15 +361,154 @@ mod tests {
         );
     }
 
+    #[rstest]
+    #[trace]
+    #[case(Seed::from_entropy())]
+    fn calculate_staker_reward_test_v1(#[case] seed: Seed) {
+        let mut rng = make_seedable_rng(seed);
+
+        let pool_id = new_pool_id(1);
+        let reward = Amount::from_atoms(rng.gen_range(1..=100_000_000));
+        let pool_balance = Amount::from_atoms(rng.gen_range(1..=100_000_000));
+        let staker_balance = Amount::from_atoms(rng.gen_range(1..=pool_balance.into_atoms()));
+
+        let cost_per_block = Amount::from_atoms(rng.gen_range(1..=reward.into_atoms()));
+        let cost_per_block_over_reward =
+            (reward + Amount::from_atoms(rng.gen_range(1..=100_000_000))).unwrap();
+        let mpt = PerThousand::new_from_rng(&mut rng);
+        let mpt_zero = PerThousand::new(0).unwrap();
+        let mpt_more_than_one = PerThousand::new(rng.gen_range(2..=1000)).unwrap();
+
+        assert_eq!(
+            calculate_staker_reward_v1(
+                Amount::ZERO,
+                Amount::ZERO,
+                Amount::ZERO,
+                Amount::ZERO,
+                mpt_zero,
+                pool_id
+            ),
+            Err(ConnectTransactionError::PoolBalanceIsZero(pool_id))
+        );
+        assert_eq!(
+            calculate_staker_reward_v1(
+                reward,
+                Amount::ZERO,
+                Amount::ZERO,
+                cost_per_block,
+                mpt_zero,
+                pool_id
+            ),
+            Err(ConnectTransactionError::PoolBalanceIsZero(pool_id))
+        );
+        assert!(calculate_staker_reward_v1(
+            reward,
+            pool_balance,
+            staker_balance,
+            Amount::ZERO,
+            mpt_zero,
+            pool_id
+        )
+        .is_ok());
+        assert!(calculate_staker_reward_v1(
+            reward,
+            pool_balance,
+            staker_balance,
+            Amount::ZERO,
+            mpt,
+            pool_id
+        )
+        .is_ok());
+        assert!(calculate_staker_reward_v1(
+            reward,
+            pool_balance,
+            Amount::ZERO,
+            Amount::ZERO,
+            mpt,
+            pool_id
+        )
+        .is_ok());
```

Comment:
Please test the expected value, not just `is_ok()`

Reply:
The idea was to check that if 0 passed to different arguments the result would be Ok/Err. The balances are random so to check exact values I'd have to duplicate the calculation. Also there are checks with arbitrary numbers below to pin the calculation.